algebra/src/fields/mod.rs

```diff
@@ -64,37 +65,27 @@ pub trait Field:
     + Sync
     + 'static
     + Eq
+    + One
     + Ord
     + Neg<Output = Self>
     + UniformRand
+    + Zero
     + Sized
     + Hash
     + From<u128>
     + From<u64>
     + From<u32>
     + From<u16>
     + From<u8>
-    + for<'a> Add<&'a Self, Output = Self>
+    + Add<Self, Output = Self>
     + for<'a> Sub<&'a Self, Output = Self>
-    + for<'a> Mul<&'a Self, Output = Self>
+    + Mul<Self, Output = Self>
     + for<'a> Div<&'a Self, Output = Self>
     + for<'a> AddAssign<&'a Self>
     + for<'a> SubAssign<&'a Self>
     + for<'a> MulAssign<&'a Self>
     + for<'a> DivAssign<&'a Self>
 {
-    /// Returns the zero element of the field, the additive identity.
-    fn zero() -> Self;
```

Comment:
Maybe it might make sense to keep these functions and have a default implementation that just invokes the methods from `Zero` and `One`? This would avoid having to type  `use num_traits::{Zero, One};` everywhere. (Ditto for the curve and group traits)

Reply:
Note that in many places, when we add the `use` for the trait, we remove the one to `Field`.

The risk I see in your suggestion is that the inherent method would be chosen ([in most cases](https://doc.rust-lang.org/reference/expressions/method-call-expr.html)) over the trait method, whereas the inherent method would not convey much more information. 
- is the `Zero` / `Mul` implementation still relevant then?
- is there a chance some future developer makes a modification to the trait implementation and forgets to apply it to the inherent method?

A way to not have to edit those `use` statements is maybe ... just to not use the `num_traits`.

Reply:
The primary reason I want the  `num_traits` impls stuff is for compatibility with generic code (such as the `ndarray` crate) that relies on these traits. 

To resolve both our concerns, maybe it suffices to keep the one and zero methods on the Field/Group traits, and then do a blanket impl:

```
impl<F: Field> One for F {
	fn one() -> F {
		<F as Field>::one()
	}
}
```

This prevents downstream developers from overriding the correct implementation, while still having the generic impls.

Reply:
I'd love this to be the case, but the traits `One` / `Zero` are non-local to zexe, so the blanket implementation on `Field`-bounded types is running afoul of orphan rules, I think.

Reply:
Bleh again the orphan rules lol. Ok, your current approach seems the best way to proceed; maybe we can re-export the traits from the algebra prelude to make life easier.